pypicloud/storage/azure_blob.py

```diff
@@ -139,7 +138,16 @@ def upload(self, package, datastream):
         metadata["version"] = package.version
 
         blob_client = self.container_client.get_blob_client(blob=path)
-        blob_client.upload_blob(data=datastream, metadata=metadata)
+        blob_client.set_blob_metadata(metadata=metadata)
```

Comment:
is there a test that validates whether the metadata arrives on cloud storage?

Reply:
Yes, look for the "test_upload" methods in `test_storage.py`

Reply:
looks like this PR is blocked by https://github.com/RaRe-Technologies/smart_open/issues/701

Reply:
opened https://github.com/RaRe-Technologies/smart_open/issues/702